### PR TITLE
Add CDTDatastore tests to build script

### PR DESCRIPTION
**Pull request by rhyshort**
Added ability to run tests from the CDTDatastore workspace, in addition
to replication acceptance tests.

ReplicationAcceptance tests are only run when -nightly true is passed
into the build script. if -nightly false is passed in or nothing is
passed in the build script will only run the tests from CDTDatastore

reviewer: @tomblench 
reviewer: @mikerhodes

### CODE REVIEW

**Comment by mikerhodes** [comment]
Seems like the option should be "-runReplicationAcceptance" rather than "-nightly". We could, I guess, have -nightly imply -runReplicationAcceptance, but they should be separate concepts.

**Comment by rhyshort** [comment]
Maybe have both? -nightly as a shorthand for -runReplicationAcceptance? The short -nightly makes the jenkins config a little easier to read on the nightly builds

File: build.rb [hunk 1]
```diff
@@ -86,9 +88,20 @@
 end
 
 if params["platform"] == "OSX"
-	system("xcodebuild -workspace ./ReplicationAcceptance/ReplicationAcceptance.xcworkspace -scheme 'RA_Tests_OSX' -destination 'platform=OS X' #{replication_options.join(" ")} test | xcpretty -r junit; exit ${PIPESTATUS[0]}")
+
+	system("xcodebuild -workspace CDTDatastore.xcworkspace -scheme 'Tests OSX' -destination 'platform=OS X' #{replication_options.join(" ")} test | xcpretty -r junit ; exit ${PIPESTATUS[0]}")
+
+	if(params["nightly"] == "true")
+		FileUtils.mv("build/reports/junit.xml","build/reports/osx.xml")
```
Comment (tomblench) [inline build.rb]: I think you should have a FileUtils.mv after each system() call which does a test run.

Maybe the names could be $params["platform"]_ra.xml and $params["platform"]_unit.xml to distinguish replication acceptance results from unit test results.

**Comment by tomblench** [comment]
I agree wtih @mikerhodes that RA or whatever makes more sense to me than nightly. I wouldn't like to see multiple synonyms for the same thing because that just gets more confusing.

Is there any use case where you would want to run RA but not unit tests?

File: build.rb (hunk 1 above)
Comment (rhyshort) [inline build.rb]: fixed in 78bfd89

**Comment by rhyshort** [comment]
okay, I've changed to option to from `nightly` to `ra`

File: build.rb [hunk 2]
```diff
@@ -6,8 +6,9 @@
 #  -platform-version iOS version to test agasint (only supported for iOS)
 #  -hardware the simultated hardware to test against eg iPhone 4S (only supported for iOS)
 #  -D* gets passed into build.
-#  
+#  -ra true | false  default is false flags if
```
Comment (tomblench) [inline build.rb]: fix this comment - it doesn't tell us what the flag does

Comment (rhyshort) [inline build.rb]: fixed in 9d3c00b

File: build.rb [hunk 3]
```diff
@@ -6,8 +6,10 @@
 #  -platform-version iOS version to test agasint (only supported for iOS)
 #  -hardware the simultated hardware to test against eg iPhone 4S (only supported for iOS)
 #  -D* gets passed into build.
-#  
+#  -ra true | false  default is false flags if true it runs the replication acceptance tests
+# 					 in addition to the normal unit tests
```
Comment (tomblench) [inline build.rb]: swap hard tabs for spaces

Comment (rhyshort) [inline build.rb]: done 7b3caeb

**Comment by tomblench** [comment]
OK +1

**Comment by mikerhodes** [comment]
+1 looks good. Are there docs for the options somewhere?

**Comment by rhyshort** [comment]
Not yet, my current plan is to have them both in the cloudant-sync repo, it'll include set up instructions for CI nodes too

**Comment by rhyshort** [comment]
Travis is having intermittent errors, one of the builds through it was fine, going to merge anyway.

**Pull request merged**